### PR TITLE
style(dashboard): Refine primary header hover and focus states, add inbox animation

### PR DESCRIPTION
**Pull request by rifont**
### What changed? Why was the change needed?
* Header updates
  * Add consistent hover and focus states for all headers (Help, Inbox, Profile)
  * Add tooltips for Help and Inbox headers (inspired by Stripe)
* Inbox updates
  * Add correct inbox bell icon 
  * Add API + Websocket backend environment variables to Inbox 
  * Add inbox bell jingle animation when new notification is received
  * Add inbox bell pulse animation when unread notifications are present
  * Show count of unread notifications on the Inbox tooltip
* Tooltip updates
  * Refactor `TooltipProvider` to App entry point
  * Set `delayDuration` to 100ms to set tooltip delay time everywhere

### Screenshots
_Header demonstration of hover, focus, and animations_

https://github.com/user-attachments/assets/c5f3c39c-f0d0-41a7-9e37-1dbc8d956c44

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

**Review by SokratisVidros — APPROVED** [review]

File: apps/dashboard/src/components/icons/inbox-bell-filled.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+
+export function InboxBellFilled(props: React.ComponentPropsWithoutRef<'svg'>) {
```
Comment (desiprisg) [inline apps/dashboard/src/components/icons/inbox-bell-filled.tsx]: Didn't even know this existed. Why not `HTMLAttributes<HTMLSvgElement>` ?

File: apps/dashboard/src/components/inbox-button.css
Comment (desiprisg) [inline apps/dashboard/src/components/inbox-button.css:1]: Love this! Can we move it to the tailwind config instead though ?

File: apps/dashboard/src/components/icons/inbox-bell-filled.tsx (hunk 1 above)
Comment (rifont) [reply apps/dashboard/src/components/icons/inbox-bell-filled.tsx]: Just copied from elsewhere, seems there's a bit of a mixture throughout the icons folder. I updated to your suggestion, it's more pure and accurate for this case.

File: apps/dashboard/src/components/inbox-button.css
Comment (rifont) [reply apps/dashboard/src/components/inbox-button.css:1]: Yes! Done 👍

**Pull request merged**